src/modules/kvs/test/cache.c

```diff
@@ -222,20 +215,13 @@ void cache_entry_treeobj_tests (void)
     ok (cache_entry_set_treeobj (e, o1) == 0,
         "cache_entry_set_treeobj success");
 
-    /* cache_entry_set_treeobj will json_decref o2 */
     ok (cache_entry_set_treeobj (e, o2) == 0,
         "cache_entry_set_treeobj again, silent success");
-    o2 = NULL;
 
     ok (cache_entry_get_valid (e) == true,
         "cache entry now valid after cache_entry_set_treeobj call");
 
-    /* cache_entry_set_raw will free data */
-    ok (cache_entry_set_raw (e, data, 4) == 0,
-        "cache_entry_set_raw, silent success");
-    data = NULL;
-
-    ok (cache_entry_set_raw (e, NULL, 0) < 0
+    ok (cache_entry_set_raw (e, data, 4) < 0
```

Comment:
actually, I think we want to keep the ```cache_entry_set_raw (e, NULL, 0)``` test, and also add ```cache_entry_set_raw (e, data, 4)``` as a new test to detect EBADE.  Both are good corner case catches.